internal/provider/resource_id_test.go

```diff
@@ -24,24 +23,36 @@ func TestAccResourceID(t *testing.T) {
 				Config: testAccResourceIDConfig,
 				Check: resource.ComposeTestCheckFunc(
 					testAccResourceIDCheck("random_id.foo", &idLens{
-						b64Len:    6,
 						b64UrlLen: 6,
 						b64StdLen: 8,
 						hexLen:    8,
 					}),
-					testAccResourceIDCheck("random_id.bar", &idLens{
-						b64Len:    12,
-						b64UrlLen: 12,
-						b64StdLen: 14,
-						hexLen:    14,
-					}),
 				),
 			},
 			{
 				ResourceName:      "random_id.foo",
 				ImportState:       true,
 				ImportStateVerify: true,
 			},
+		},
+	})
+}
+
+func TestAccResourceID_importWithPrefix(t *testing.T) {
+	resource.UnitTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceIDConfigWithPrefix,
```

Comment:
Would it make sense to move all those strings inline?

I feel like our tests are already an articulate structure, as `resource.Test()` require it. Whenever possible, having `Config` string defined there and then, imho, makes for easier reading (i.e. I don't have to scroll up and down to find what I'm testing and see how I'm testing it).

The fact that you are already splitting those strings into separate resources does help a lot already.

Reply:
As some of the same `Config` strings are used in multiple locations I think I'm going to leave these as is for now. Happy to revisit at a later date if it seems like inlining would be better.